### PR TITLE
Paweljacyno dev

### PR DESCRIPTION
**Pull request by paweljacyno**

### CODE REVIEW

**Review by maxster256 — COMMENTED** [review]
W twojej pracy najbardziej podoba mi się wybór regresji logistycznej jako jednego z modeli, jako że nie był on za często wybierany, a u Ciebie daje to ciekawe wyniki. Z minusów no to imo brakuje trochę dogłębnej analizy użytych modeli. W większości z nich badana jest tylko jedna metryka - dokładność, co nie daje pełnego obrazu jakości modelu. Ponadto poleceniem było zbadanie zależności pomiędzy metrykami a hiperparametrami, czego tutaj również brak.

**Review by xCookieMonster05x — COMMENTED** [review]
Spoko, że rozważyłeś różne modele i bardzo zwięźle. Moim zdaniem mógłbyś trochę więcej napisać/opisać i zwizualizować. Kod jak najbardziej poprawny :)

**Comment by slovik02** [comment]
Twoja praca jest na pewno zwięzła i jak ktoś już wspomniał wcześniej, fajnie jest zobaczyć, że ktoś spróbował zrobić regresję liniową :)
Widać że spoko rozumiesz podstawy i super, że starałeś się też samodzielnie kombinować z hiperparametrami. 
Użycie Dummy Classifiera też poprawne.

Na plus też classification report, ale zabrakło komentarzy do wyników, np. że model lepiej radzi sobie z klasą 0 niż 1, co wyraźnie widać po recall i precision. 

Podział danych można by zapisać czytelniej, używając konstrukcji (test size i random state możesz zmienić wedle uznania, ale w granicy rozsądku): 

X=dane.drop('Survived', axis=1)
y=dane['Survived']
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

dzięki temu od razu jasno widać, które dane trafiają do treningu, a które do testów. 

Mam też wrażenie, że w kilku miejscach oceniasz model na tych samych danych, na których go wytrenowałeś, co nie daje rzetelnej informacji o jego jakości. Dużo ważniejsze są wyniki na danych testowych. Taki jakby egzamin, który zrobiłeś ale dla każdego modelu. 

Aby skutecznie też znaleźć najbardziej optymalne hiperparametry możesz sobie sprawdzić GridSearchCV : )

**Review by Oskipol — COMMENTED** [review]
Spoko, że porównałeś różne modele i użyłeś DummyClassifiera jako baseline. Brakuje mi trochę szerszej analizy wyników (precision, recall, F1) i komentarza do różnic między modelami, ale całość jest czytelna i poprawna.
